### PR TITLE
feat(sweeps): Sweep now has an expected_run_count property 

### PR DESCRIPTION
**Pull request by gtarpenning**
[11226](https://wandb.atlassian.net/jira/software/c/projects/WB/boards/62?modal=detail&selectedIssue=WB-11226)

Description
-----------
Adds an expected_run_count attribute to the sweeps object, which is the number of expected sweeps in a grid search. This just = the Cartesian product of the number of params. 

Testing
-------
`test_public_api.py::test_sweep_api_expected_run_count`

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/master/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

**Comment by gtarpenning** [comment]
One caveat to this PR is that the `dummySweepOrchestrator` in `core` doesn't handle nested parameters, whereas anaconda does. This leads to the nested param test failing (here) in `wandb`, but working in real environments that hit anaconda. 2 reasons this is fine: 
1. The `core` side of this PR is already in, and passes the anaconda tests. 
2. Nested parameters are very niche, the vast majority of sweeps use normal params.

File: wandb/apis/public.py [hunk 1]
```diff
@@ -2565,6 +2567,14 @@ def best_run(self, order=None):
         except IndexError:
             return None
 
+    @property
+    def expected_run_count(self) -> int:
+        "Returns the number of expected runs in the sweep or -1 for infinite runs."
+        expected_run_count = self._attrs.get("runCountExpected", -1)
+        if expected_run_count is None:
+            expected_run_count = -1
```
[the remark this answers is not in the document]
Comment (KyleGoyette) [reply wandb/apis/public.py:2575]: I feel like I would be inclined to make this return an optional int, where None means there is no expected count.

**Review by KyleGoyette — COMMENTED** [review]
I think the public api should return None if the estimated run count doesn't exist for that optimizer

**Review by KyleGoyette — APPROVED** [review]
🚢

**Pull request merged**